### ISSUE TITLE
Option to disable autochange country when typing

### ISSUE THREAD

**Issue by Styort** [cedvdb/phone_form_field#269]
I have set up a list of supported countries in the `CountrySelectorNavigator` (by passing a `List<IsoCode>` to `countries` parameter of `CountrySelectorNavigator `constructor ) and want to be able to enter a phone number for supported countries only. But if I start entering the phone number prefix of another country that I do not have in the list, it automatically changes to it.
[Here ](https://github.com/user-attachments/assets/acb89dbc-622f-45bf-af5e-60e8d42d3ff1) is a video with an example of reproducing the bug.

Is is possible to disable now?

**Comment by cedvdb** [comment]
What would be the expected behavior ?

**Comment by Styort** [comment]
> What would be the expected behavior ?
> 
> What would it display when you do +1 ?

Just continue displaying +1, without switching countries.

**Comment by cedvdb** [comment]
What do you think about this :

  - when the user uses a country that is not in the list of selected countries, show an error message ?

**Comment by Styort** [comment]
I would not like to give the opportunity to change the country if it is not on the list of available ones.

I would like to be able to strictly limit the input of unsupported numbers so that the user immediately understands that the application is not available in his country.

I thought that's what the list of `countries` in `CountrySelectorNavigator` was for...

**Comment by Styort** [comment]
Any updates on this issue?
Are there plans to implement this feature? If yes, approximately when?

**Comment by cedvdb** [comment]
> Just continue displaying +1, without switching countries.

Again I don't think this is acceptable. It should change the country with an invalid message and not display the country in the list of countries, imo.

> Are there plans to implement this feature? If yes, approximately when?

Can you implement it ? Let's land on a design for the first point of this comment first though.

**Comment by Styort** [comment]
> I'm not sure about this, it seems counter intuitive from an UX standpoint.

What then is the point of the `countries` parameter in the `CountrySelectorNavigator`? After all, in the end, no matter what `countries` are in the `CountrySelectorNavigator`, I can select any one by writing its prefix. This behavior is confusing...

If you strictly prohibit changing the country when typing a prefix, if it is not in the list of allowed ones, then this can become breaking changes, which is not good. Therefore, I would suggest introducing a special flag for this. For example `allowOnlySelectorNavigatorCountries` or `restrictCountriesByCountrySelectorNavigator` (or you can suggest your own name) the default value of which will be `false`.

> Can you implement it ?

I can try it when I have free time (maybe this month).

**Comment by cedvdb** [comment]
> What then is the point of the countries parameter in the CountrySelectorNavigator

It does not let the option to the user to select that country in the list

 >  I can select any one by writing its prefix
 
 Yes currently that's how it works, and it's not perfect. I'm still of the opinion that just writing `+1` and not changing the country without any feedback to the user is a worst solution. If the input displayed an error message below like "This country is not allowed", at least there is a feedback, otherwise the user may conclude that typing a + does not work.

**Comment by Styort** [comment]
> I'm still of the opinion that just writing `+1` and not changing the country without any feedback to the user is a worst solution.

I agree that displaying an error is a good option, but I wouldn’t want the country’s flag to change at the same time, and would simply leave the input field as is, but with an input error.

**Comment by cedvdb** [comment]
Let me explain my rationale:

We both agree that feedback should be conveyed to the user when he tries to enter an invalid country, an error message should therefor be shown, we agree on that part.

If that's the case, then the input is going to be in an invalid state whether or not the flag is changed.

If we do not change the flag:

  - The user may get confused and conclude that typing +"country code" doesn't work at all.
  - In the scenario where the user is not 100% sure about the country code this could lead to confusion. For example if an user  types a phone number from his friend in France but for some reason thinks that the country dial code of france is +34, showing a spanish flag will be helpful (even if that flag is invalid).
  - I do not see any added benefit of not including the flag change when there is an error message shown

**Comment by cedvdb** [comment]
Maybe I misinterpreted your original comment, you want to disable it for all countries or only invalid countries ?

**Comment by KaiExa** [comment]
Hi is it done added? for this feature
